[WIP] Rename from API to Http-Client

AliabbasMerchant opened the pull request:
I'll keep on updating `renaming_status.txt`.  
The file will be deleted before the final merge.   

(The module may be unusable while the process of renaming is going on.)

alexander255 commented:
Thanks for working on this again!
One comment after looking over the first version: Do you really want to rename the root package from `ipfsapi` to `ipfs_http_client`? Why not use `ipfshttpclient` instead, as it is standard python convention?

AliabbasMerchant commented:
@daviddias @alexander255  
I have made all changes from my side.  
Please have a look at `renaming_status.txt`  
I have marked a few things as doubtful

alexander255 commented:
Looks mostly good. Please take a look at those CI failures (aside from codestyle). Also please don't create the `-ipfshttpclient.*` files – just leave them at their original name. Once I merge this I'll take care to split off the previous version into another branch and start working on that compat-shim I talked about.

AliabbasMerchant commented:
1. I'll look into the CI failure.  
2. I havent renamed any of the files in `ipfshttpclient`. Just changed the folder name from `ipfsapi` to `ipfshttpclient`, and changed the contents of the files.

alexander255 commented:
> I havent renamed any of the files in ipfshttpclient. Just changed the folder name from ipfsapi to ipfshttpclient, and changed the contents of the files.

Then please delete the original files (`git rm`), commit that and [squash your commits](https://stackoverflow.com/a/5201642/277882). GIT will then detect the files as “renamed with changes” and `git blame` will continue to work. As I said please also rename the `-ipfshttpapi.*` files back to their original names.

Thanks for taking the time to work on this and all the work you've already submitted thus far!

AliabbasMerchant commented:
@alexander255 I have renamed the 3 `-ipfshttpclient.*` files back to their original name and have made changes in them, under the original name itself.  
Also, I have squashed all the commits.  
With regards to `docs/api_ref.md` -> `docs/http_client_ref.md`, no matter what I do, git is considering it as a delete-new_file and not a rename.

alexander255 commented on renaming_status.txt:
```diff
@@ -0,0 +1,81 @@
+ipfshttpclient/encoding.py        Ln 392
+ipfshttpclient/http.py            Ln 2,4,134
+ipfshttpclient/version.py         Ln 7
+test/run-tests.py                 Ln 68,102
```
Reviewed all these and all, except for the ones in `client.py`, indeed refer to the IPFS HTTP API and should be changed. Current state looks good.

alexander255 commented on renaming_status.txt:
```diff
@@ -0,0 +1,81 @@
+FOR README.md:
+* I have made changes in the original file itself
+* External changes to be done by the project maintainers (I have already made the changes in the file, on the respective lines):
+    https://img.shields.io/pypi/v/ipfsapi.svg?style=flat-square -> https://img.shields.io/pypi/v/ipfshttpclient.svg?style=flat-square   line 6
```
Will happen automatically when we push the new PyPI package.

alexander255 commented on renaming_status.txt:
```diff
@@ -0,0 +1,81 @@
+* I have made changes in the original file itself
+* External changes to be done by the project maintainers (I have already made the changes in the file, on the respective lines):
+    https://img.shields.io/pypi/v/ipfsapi.svg?style=flat-square -> https://img.shields.io/pypi/v/ipfshttpclient.svg?style=flat-square   line 6
+    https://pypi.python.org/pypi/ipfsapi -> https://pypi.python.org/pypi/ipfshttpclient                                                 line 6
```
Will need to be manually just after this branch is merged. Needs transitional code for old package in place.

alexander255 commented on renaming_status.txt:
```diff
@@ -0,0 +1,81 @@
+    https://img.shields.io/pypi/v/ipfsapi.svg?style=flat-square -> https://img.shields.io/pypi/v/ipfshttpclient.svg?style=flat-square   line 6
+    https://pypi.python.org/pypi/ipfsapi -> https://pypi.python.org/pypi/ipfshttpclient                                                 line 6
+    https://travis-ci.org/ipfs/py-ipfs-api.svg?branch=master -> https://travis-ci.org/ipfs/py-ipfs-http-client.svg?branch=master        line 7
+    https://travis-ci.org/ipfs/py-ipfs-api -> https://travis-ci.org/ipfs/py-ipfs-http-client                                            line 7
```
Will happen automatically once the source repo is renamed.

alexander255 commented on renaming_status.txt:
```diff
@@ -0,0 +1,81 @@
+    https://travis-ci.org/ipfs/py-ipfs-api.svg?branch=master -> https://travis-ci.org/ipfs/py-ipfs-http-client.svg?branch=master        line 7
+    https://travis-ci.org/ipfs/py-ipfs-api -> https://travis-ci.org/ipfs/py-ipfs-http-client                                            line 7
+    https://ipfs.io/ipns/QmZ86ow1byeyhNRJEatWxGPJKcnQKG7s51MtbHdxxUddTH/Software/Python/ipfsapi/ ->
+                    https://ipfs.io/ipns/QmZ86ow1byeyhNRJEatWxGPJKcnQKG7s51MtbHdxxUddTH/Software/Python/ipfshttpclient/                 line 11,118
```
Needs docs to be regenerated and pushed by me.

alexander255 commented on renaming_status.txt:
```diff
@@ -0,0 +1,81 @@
+    https://travis-ci.org/ipfs/py-ipfs-api -> https://travis-ci.org/ipfs/py-ipfs-http-client                                            line 7
+    https://ipfs.io/ipns/QmZ86ow1byeyhNRJEatWxGPJKcnQKG7s51MtbHdxxUddTH/Software/Python/ipfsapi/ ->
+                    https://ipfs.io/ipns/QmZ86ow1byeyhNRJEatWxGPJKcnQKG7s51MtbHdxxUddTH/Software/Python/ipfshttpclient/                 line 11,118
+    The PIP package and Python module: ipfshttpclient
```
Needs transitional module for old name in-place.

alexander255 commented on renaming_status.txt:
```diff
@@ -0,0 +1,81 @@
+    https://ipfs.io/ipns/QmZ86ow1byeyhNRJEatWxGPJKcnQKG7s51MtbHdxxUddTH/Software/Python/ipfsapi/ ->
+                    https://ipfs.io/ipns/QmZ86ow1byeyhNRJEatWxGPJKcnQKG7s51MtbHdxxUddTH/Software/Python/ipfshttpclient/                 line 11,118
+    The PIP package and Python module: ipfshttpclient
+    The Github repo: py-ipfs-api -> py-ipfs-http-client
```
Needs help from org maintainers iirc.

alexander255 commented on renaming_status.txt:
```diff
@@ -0,0 +1,81 @@
+                    https://ipfs.io/ipns/QmZ86ow1byeyhNRJEatWxGPJKcnQKG7s51MtbHdxxUddTH/Software/Python/ipfshttpclient/                 line 11,118
+    The PIP package and Python module: ipfshttpclient
+    The Github repo: py-ipfs-api -> py-ipfs-http-client
+    PROJECT RENAME:        Python IPFS API -> Python IPFS HTTP Client
```
Happens automatically when this is merged.

alexander255 commented on test/unit/test_http.py:
```diff
@@ -219,18 +219,19 @@ def test_session(self):
 def test_stream_close(mocker):
-    client = ipfsapi.http.HTTPClient("localhost", 5001, "api/v0")
-    mocker.patch("ipfsapi.http._notify_stream_iter_closed")
+    client = ipfshttpclient.http.HTTPClient("localhost", 5001, "http_client/v0")
```
This is refering to the Go-IPFS API endpoint. You cannot change this.

alexander255 commented on test/unit/test_http.py:
```diff
@@ -139,10 +139,10 @@ class TestHttp(unittest.TestCase):
             'localhost',
             5001,
-            'api/v0')
+            'http_client/v0')
```
This is refering to the Go-IPFS API endpoint. You cannot change this.

AliabbasMerchant commented:
Ok. I'll do the required changes. Just give me a few days time. I am currently not at home and don't have my laptop. :)

AliabbasMerchant commented:
Finally! No conflicts.  
The file `api_ref.md` was successfully renamed to `http_client_ref.md`  
Also, I have changed `http_client/v0` to `api/v0`

alexander255 commented on test/unit/test_http.py:
```diff
@@ -170,45 +170,45 @@ def test_stream(self):
-        with HTTMock(api_okay):
-            res = self.client.request('/apiokay')
+        with HTTMock(http_client_okay):
+            res = self.client.request('/http_client_fail')
```
This should be `/http_client_okay`

alexander255 commented on test/unit/test_http.py:
```diff
@@ -170,45 +170,45 @@ def test_stream(self):
-        with HTTMock(api_okay):
-            res = self.client.request('/apiokay',
+        with HTTMock(http_client_okay):
+            res = self.client.request('/http_client_fail',
```
Same here
